### PR TITLE
add osfclient

### PR DESCRIPTION
**Pull request by untzag**

### CODE REVIEW

File: recipes/osfclient/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 1
```
Comment (dougalsutherland) [inline recipes/osfclient/meta.yaml]: Should start at 0. :)

File: recipes/osfclient/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{% set name = "osfclient" %}
+{% set version = "0.0.3" %}
+{% set sha256 = "26a0d8f5611f50b8e3ff37a0ce37959e77fbc6941d180aecd9b4242e75d69823" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (dougalsutherland) [inline recipes/osfclient/meta.yaml]: I know it's in the template, but we don't actually need this line.

File: recipes/osfclient/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 1
+  noarch: python
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
```
Comment (dougalsutherland) [inline recipes/osfclient/meta.yaml]: You should add [entry points](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points) here to match [the ones in `setup.py`](https://github.com/dib-lab/osf-cli/blob/0.0.3/setup.py#L52-L56). (Unfortunately you have to specify them in the recipe too.)

File: recipes/osfclient/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  run:
+    - python
+    - requests
+    - tqdm
+    - six
+
+test:
+  imports:
+    - osfclient
+
```
Comment (dougalsutherland) [inline recipes/osfclient/meta.yaml]: Test the `osf` script here too (just `osf --help` should be fine).

**Comment by dougalsutherland** [comment]
Thanks @untzag! Few minor things to address. It's possible the entry_point stuff won't work with noarch, but let's see first.

**Comment by untzag** [comment]
thanks so much for the pointers @dougalsutherland

**Comment by dougalsutherland** [comment]
Oh, one more minor thing and then it looks good:

You should for legal reasons include a `license_file`. Unfortunately, the license isn't currently in the distributions of the package; I just sent a PR to fix that (https://github.com/dib-lab/osf-cli/pull/112). In the meantime, you can [package it manually](http://conda-forge.readthedocs.io/en/latest/meta.html#packaging-the-license-manually) (and put a comment linking to the PR as a reminder that it can hopefully be deleted for future releases).

**Comment by dougalsutherland** [comment]
Done. Thanks @untzag!

**Pull request merged**